### ISSUE TITLE
Accessibility issues on the community-list page

### ISSUE THREAD

**Issue by doana** [DSpace/dspace-angular#2251]
**Describe the bug**
The community list page (/community-list) has some serious challenges regarding accessibility. 

1. Improper heading structure: The issue that drew my attention initially was that the headings are not properly nested. The page lacks a h1 at all. The title at the top of the page is an h2, and then each community on the page is a h5.
2. Keyboard navigation: Expanding any of the communities on the page resets the focus back up to the top of the list. When this happens there is no visual indicator of where the focus has gone.
3. Screen reader support. I tested the page in the NVDA screen reader and ran across a number of issues. 
    1. Telling the software to read the entire contents of the page reads only the first item in the list and then skips to the bottom of the page.
    2. The expand / collapse buttons aren't consistently read out loud. It seems like it will read the first one but doesn't announce the others at all.
    3. The nested structure of the list isn't communicated at all.

**To Reproduce**
1. Visit the community list page
2. Try to interact with the community list using keyboard controls only, tabbing through the list and using the space / enter keys to expand communities
3. Try to read the full community list aloud with the NVDA screen reader
4. Try to interact with the community list controls using the NVDA screen reader

**Expected behavior**
- If you're going to use headings for the community / collection names at all (I don't think it's really necessary), then they should be hierarchical starting with h1 and then descending down the heading order as necessary, with h2s for each top level community, h3's below each of those and so on. 
- You should be able to expand and collapse the community tree without having the focus reset to the top of the list each time. 
- The focus indicator should always be visible when navigating the page via the keyboard
- If you read the page in its entirety using a screen reader, you should be able to hear all of the page content
- The community / collection hierarchy should be communicated to screen reader users
 
To my eyes, it seems like the angular cdk-tree component is likely the cause of many of these issues. A lot of these challenges would be resolved if the collection structure were just rendered with simple nested unordered lists.

**Comment by tdonohue** [comment]
Agreed with your overall analysis here @doana .  This relates back to an older ticket regarding accessibility issues on this `/community-list` page: #1177  

It's likely we need to find someone to help refactor the code at some point to no longer use `cdk-tree` (https://material.angular.io/cdk/tree/overview) or see if there is a way to improve accessibility while using it.
